objects/src/testing/account_id.rs

```diff
@@ -120,3 +122,89 @@ pub const fn account_id<const CHECK_MIN_ONES: bool>(
+/// Refer to its documentation for details.
+///
+/// # Example
+///
+/// ```
+/// # use miden_objects::accounts::{AccountType, AccountStorageMode, AccountId};
+/// # use miden_objects::testing::account_id::{AccountIdBuilder};
+///
+/// let mut rng = rand::thread_rng();
+///
+/// // A random AccountId with random AccountType and AccountStorageMode.
+/// let random_id1: AccountId = AccountIdBuilder::new().build_with_rng(&mut rng);
+///
+/// // A random AccountId with the given AccountType and AccountStorageMode.
+/// let random_id2: AccountId = AccountIdBuilder::new()
+///     .account_type(AccountType::FungibleFaucet)
+///     .storage_mode(AccountStorageMode::Public)
+///     .build_with_rng(&mut rng);
+/// assert_eq!(random_id2.account_type(), AccountType::FungibleFaucet);
+/// assert_eq!(random_id2.storage_mode(), AccountStorageMode::Public);
+/// ```
+pub struct AccountIdBuilder {
+    account_type: Option<AccountType>,
+    storage_mode: Option<AccountStorageMode>,
+}
+
+impl AccountIdBuilder {
+    /// Creates a new [`AccountIdBuilder`].
+    pub fn new() -> Self {
+        Self { account_type: None, storage_mode: None }
+    }
+
+    /// Sets the [`AccountType`] of the generated [`AccountId`] to the provided value.
+    pub fn account_type(mut self, account_type: AccountType) -> Self {
+        self.account_type = Some(account_type);
+        self
+    }
+
+    /// Sets the [`AccountStorageMode`] of the generated [`AccountId`] to the provided value.
+    pub fn storage_mode(mut self, storage_mode: AccountStorageMode) -> Self {
+        self.storage_mode = Some(storage_mode);
+        self
+    }
+
+    /// Builds an [`AccountId`] using the provided [`rand::Rng`].
+    ///
+    /// If no [`AccountType`] or [`AccountStorageMode`] were previously set, random ones are
+    /// generated.
+    pub fn build_with_rng<R: rand::Rng + ?Sized>(self, rng: &mut R) -> AccountId {
+        let account_type = match self.account_type {
+            Some(account_type) => account_type,
+            None => rng.gen(),
+        };
+
+        let storage_mode = match self.storage_mode {
+            Some(storage_mode) => storage_mode,
+            None => rng.gen(),
+        };
+
+        AccountId::dummy(rng.gen(), account_type, storage_mode)
+    }
+
+    /// Builds an [`AccountId`] using the provided seed as input for a [`rand::rngs::StdRng`].
+    ///
+    /// If no [`AccountType`] or [`AccountStorageMode`] were previously set, random ones are
+    /// generated.
+    pub fn build_with_seed(self, rng_seed: [u8; 32]) -> AccountId {
+        // On non-64 bit architectures, SmallRng expects 16 bytes as seed input.
+        #[cfg(not(target_pointer_width = "64"))]
+        let rng_seed: [u8; 16] = rng_seed.as_slice()[0..16]
+            .try_into()
+            .expect("we should have sliced off 16 elements");
+        let mut rng = rand::rngs::SmallRng::from_seed(rng_seed);
+        self.build_with_rng(&mut rng)
+    }
```

Comment:
It's nice to be able to use the Rng API, so I'm using the seed to create an RNG.

However for `SmallRng`:
> The algorithm is deterministic but should not be considered reproducible due to dependence on platform and possible replacement in future library versions. For a reproducible generator, use a named PRNG from an external crate, e.g. [rand_xoshiro](https://crates.io/crates/rand_xoshiro) or [rand_chacha](https://crates.io/crates/rand_chacha). Refer also to [The Book](https://rust-random.github.io/book/guide-rngs.html).

So maybe we should use `rand_xoshiro` (which is what `SmallRng` currently uses) so tests don't break in the future?

Reply:
As an aside why use `SmallRng` over `StdRng`? But probably we should just use the given crates directly.